bigtop-deploy/puppet/manifests/python.pp

```diff
@@ -14,16 +14,70 @@
 # limitations under the License.
 
 class python {
-    if (($operatingsystem == 'Ubuntu' and 0 <= versioncmp($operatingsystemmajrelease, '22.04'))) {
-        file { '/usr/bin/python':
-          ensure => 'link',
-          target => '/usr/bin/python2',
+
+    define set_python_alternatives($command) {
+        exec { $name:
+          command => $command,
+          unless  => "/usr/sbin/update-alternatives --display python | grep '/usr/bin/python3'",
+          path    => ['/bin', '/usr/bin', '/sbin', '/usr/sbin'],
         }
     }
-    
-    if ($operatingsystem == 'openEuler') {
-        package { 'python3-unversioned-command':
-          ensure => 'present'
+
+    define change_python_interpreter($file, $interpreter) {
+        file_line { "change_interpreter_in_${name}":
+          path  => $file,
+          line  => "#!${interpreter}",
+          match => '^#!.*python.*$',
+        }
+    }
+
+    case $operatingsystem {
+        /(?i:(centos|fedora|redhat|rocky))/: {
+            if ($operatingsystem != 'Fedora') {
+                case $operatingsystemmajrelease {
+                    '9': {
+                        package { 'python-unversioned-command':
+                          ensure => 'present',
+                        }
+                    }
+                    '8': {
+                        set_python_alternatives { 'set-python3':
+                          command => '/usr/sbin/update-alternatives --set python /usr/bin/python3',
+                        }
+                    }
+                    '7': {
+                        change_python_interpreter { 'yum':
+                          file        => '/usr/bin/yum',
+                          interpreter => '/usr/bin/python2.7',
+                        }
+                        change_python_interpreter { 'urlgrabber-ext-down':
+                          file        => '/usr/libexec/urlgrabber-ext-down',
+                          interpreter => '/usr/bin/python2.7',
+                        }
```

Comment:
We seem not be able to cover all scenario by this hack.

```
Debug: Executing '/usr/bin/python /usr/local/share/gems/gems/puppet-3.6.2/lib/puppet/provider/package/yumhelper.py'
Error: Could not get latest version: Execution of '/usr/bin/python /usr/local/share/gems/gems/puppet-3.6.2/lib/puppet/provider/package/yumhelper.py' returned 1: File "/usr/local/share/gems/gems/puppet-3.6.2/lib/puppet/provider/package/yumhelper.py", line 40
    raise ValueError, a
                    ^
SyntaxError: invalid syntax
Error: /Stage[main]/Bigtop_utils::Client/Package[bigtop-utils]/ensure: change from 3.3.0-1.el7 to latest failed: Could not get latest version: Execution of '/usr/bin/python /usr/local/share/gems/gems/puppet-3.6.2/lib/puppet/provider/package/yumhelper.py' returned 1: File "/usr/local/share/gems/gems\
/puppet-3.6.2/lib/puppet/provider/package/yumhelper.py", line 40
    raise ValueError, a
                    ^
SyntaxError: invalid syntax
```

Reply:
Alright, we'll have to temporarily ignore dealing with CentOS 7 for now.